Assets/Scripts/DrawPlanetToStarBeam.cs

```diff
@@ -0,0 +1,118 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+// when hover, 
+public class DrawPlanetToStarBeam : MonoBehaviour
+{
+    private LineRenderer beam;
+    private float counter;
+    private float dist;
+
+    public Transform player;
+
+    public Transform curPlanet;
+    public Transform destinStar;
+
+    public float beamDrawSpeed;
+    public float normalBeamWidth = 30.0f;
+    //public float mapClearBeamWidth = 20.0f;
+
+    private bool _levelClear;
+    private bool _connectBeam;
+    private bool _draw;
+
+    // Start is called before the first frame update
+    private void Start()
+    {
+        StarIconManager.OnHoverStart += OnHoverStart;
+        StarIconManager.OnHoverStop += OnHoverStop;
+        StarIconManager.OnSelectStar += OnSelectStar;
+        // _connectBeam = false;
+        player = transform.parent;
+
+    }
+
+    private void OnSelectStar(Transform obj)
+    {
+        _draw = false;
+    }
+    
+    private void OnHoverStart(Transform targetStar)
+    {
+        print("on hover star -- draw planet to star beam");
+        print(targetStar.name);
+
+        destinStar = targetStar;
+        // planet core parent
+
+        player = transform.parent;
+        if (!player.parent) return;
+        
+        print("on hover star --" + player.parent.parent.name);
+        
+        if (!player.parent.parent.CompareTag("|Planet|")) return;
+        print("on hover star -- draw planet to star beam -- find planet");
+        curPlanet = player.parent.parent;
```

Comment:
The planet core's position is always centred on the planet; wouldn't drawing a line from the core to the star be the same thing as drawing one from the planet to the star, given they're at the same location?